client/web/src/org/settings/DeleteOrgModal.tsx

```diff
@@ -0,0 +1,98 @@
+
+import CloseIcon from 'mdi-react/CloseIcon'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useHistory } from 'react-router'
+import { RouteComponentProps } from 'react-router-dom'
+
+import { asError } from '@sourcegraph/common'
+import { Button, Input, LoadingSpinner, Modal } from '@sourcegraph/wildcard'
+
+import { deleteOrganization } from '../../site-admin/backend'
+import { eventLogger } from '../../tracking/eventLogger'
+import { OrgAreaPageProps } from '../area/OrgArea'
+
+interface DeleteOrgModalProps extends OrgAreaPageProps, RouteComponentProps<{}>  {
+    isOpen: boolean
+    toggleDeleteModal: () => void
+}
+
+export const DeleteOrgModal: React.FunctionComponent<DeleteOrgModalProps> =  props => {
+    const { org, isOpen, toggleDeleteModal } = props
+    const deleteLabelId = 'deleteOrgId'
+    const [orgNameInput, setOrgNameInput] = useState('')
+    const [isOrgNameValid, setIsOrgNameValid] = useState<boolean>()
+    const [loading, setLoading] = useState<boolean | Error>(false)
+
+    const history = useHistory()
+
+    useEffect(() => { setOrgNameInput(orgNameInput) }, [setOrgNameInput, orgNameInput])
+
+    const onOrgChangeName = useCallback<React.ChangeEventHandler<HTMLInputElement>>(event => {
+        setOrgNameInput(event.currentTarget.value)
+        setIsOrgNameValid(event.currentTarget.value === org.name)
+    }, [org])
+
+    const deleteOrg = useCallback(
+        async() => {
+            setLoading(true)
+
+            try {
+                await deleteOrganization(org.id, true)
+                setLoading(false)
+                history.push({
+                    pathname: '/settings',
+                })
+
+            } catch(error)   {
+                setLoading(asError(error))
+                eventLogger.log('OrgDeletionFailed')
+            }
+        },
+        [org.id, history])
+
+    return (
+        <Modal
+            position="center"
+            isOpen={isOpen}
+            onDismiss={toggleDeleteModal}
+            aria-labelledby={deleteLabelId}
+            data-testid="delete-org-modal"
+        >
+
+        {props.org.viewerIsMember && (
+            <div>
+                <h3 className="text-danger" id={deleteLabelId}>
+                    Delete organization?
+                </h3>
+                <CloseIcon
+                    className="icon-inline position-absolute cursor-pointer"
+                    style={{ top: '1rem', right: '1rem' }}
+                    onClick={toggleDeleteModal}
+                />
+                <p className="pt-3">
+                    <strong>You are going to delete { org.name } from Sourcegraph.</strong>This cannot be undone. Deleting an organization will remove all of its synced repositories from Sourcegraph, along with the organization's code insights, batch changes, code monitors and other resources.
```

Comment:
Nit - split into multiple lines
```suggestion
                    <strong>You are going to delete { org.name } from Sourcegraph.</strong>
                    This cannot be undone. Deleting an organization will remove all of its synced
                    repositories from Sourcegraph, along with the organization's code insights, 
                    batch changes, code monitors and other resources.
```